data/src/test/java/org/oppia/android/data/backends/gae/RemoteAuthNetworkInterceptorTest.kt

```diff
@@ -92,51 +89,82 @@ class RemoteAuthNetworkInterceptorTest {
     assertThat(serviceRequest.header("app_version_code")).isNull()
 
     call.execute()
-    val interceptedRequest = mockWebServer.takeRequest(timeout = 8, unit = TimeUnit.SECONDS)
-    verifyRequestHeaders(interceptedRequest!!.headers)
+    val interceptedRequest = mockWebServer.takeRequest(
+      timeout = testCoroutineDispatcher.DEFAULT_TIMEOUT_SECONDS,
+      unit = testCoroutineDispatcher.DEFAULT_TIMEOUT_UNIT
+    )
+    interceptedRequest?.let {
+      verifyRequestHeaders(it.headers)
+    }
   }
 
   @Test
-  fun testNetworkInterceptor_withHeaders_setsCorrectHeaders() {
-    mockWebServer.enqueue(MockResponse().setBody("{}"))
-    val call = topicService.getTopicByName(topicName)
-
-    val serviceRequest = topicService.getTopicByName(topicName).request()
-    val request = serviceRequest.newBuilder()
+  fun testNetworkInterceptor_withIncorrectHeaders_setsCorrectHeaders() {
+    val request = Request.Builder()
+      .url(mockWebServer.url("/"))
```

Comment:
Maybe explain briefly via comment why we can't use Retrofit in this test case.

Reply:
Done